src/component/table/Table.jsx

```diff
@@ -0,0 +1,170 @@
+// @flow
+import React, {Component, PropTypes} from 'react';
+import {fromJS, List, Map} from 'immutable';
+import {deepReduceOutwards} from 'immutable-recursive';
+import ComponentClassName from '../../util/ComponentClassName';
+
+/**
+ * @module Table
+ */
+
+//
+// Th
+
+type ThProps = {
+    schemaItem: Map<any,any>
+}
+
+function Th(props: ThProps): React.Element<any> {
+    var {width, heading = ''} = props.schemaItem.toObject();
+    return <th
+        style={{width}}
+        children={heading}
+    />;
+}
+
+
+//
+// Td
+
+type TdProps = {
+    row: Map<any,any>,
+    schemaItem: Map<any,any>
+}
+
+function Td(props: TdProps): React.Element<any> {
+    const {row, schemaItem} = props;
+    const {render, filter} = schemaItem.toObject();
+
+    // Content rendering priority order
+    // 1. render function
+    // 2. filter function
+    // 3. filter key accessor
+    const content = (render)
+        ? render(row)
+        : (typeof filter === 'function')
+            ? filter(row)
+            : row.get(filter)
+    ;
+
+    // Only return a td if the schema hasn't
+    // provided one
+    return (content && content.type === 'td')
+        ? content
+        : <td>{content}</td>;
+}
+
+
+
+
+
+type Schema = ListOrArray<SchemaItem>;
+
+type SchemaItem = {
+    filter: string | (row: Map<any>) => React.Element<any>,
```

Comment:
Not sure about the naming of `filter` it isn't really doing filtering so much as transforming data. It isn't perfect but a possible alternative is what react-virtualized uses [`cellDataGetter`](https://github.com/bvaughn/react-virtualized/blob/master/docs/Column.md)